src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -71,6 +73,34 @@ public override void OnActionExecuted(HttpActionExecutedContext actionExecutedCo
                 throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveActionContext);
             }
 
+            bool hasExpandQueryParameter = false;
+            string queryString = request.RequestUri.Query;
+            var absoluteUri = request.RequestUri.AbsoluteUri;
+            bool queryStringIsEmpty = string.IsNullOrEmpty(queryString);
+
+            ReadOnlySpan<char> absoluteUriSpan = absoluteUri.AsSpan();
+            string baseUriWithPath = queryStringIsEmpty ? absoluteUri : absoluteUriSpan.Slice(0, absoluteUri.IndexOf('?', 0)).ToString();
+
+            if (!queryStringIsEmpty)
+            {
+                string[] queryParameters = queryString.Split('&');
+
+                hasExpandQueryParameter = queryParameters.Any(x => x.StartsWith("?$expand", StringComparison.OrdinalIgnoreCase) || x.StartsWith("$expand", StringComparison.OrdinalIgnoreCase) || x.StartsWith("expand", StringComparison.OrdinalIgnoreCase));
+            }
+
+            // Create a $expand query string if 1) It's a POST request 2) if there is no expand query string.
+            if (String.Equals(actionExecutedContext.Request.Method, HttpMethod.Post) && !hasExpandQueryParameter)
+            {
+                string expand = GenerateExpandQueryFromPayload(actionExecutedContext.ActionContext);
+
+                if (!string.IsNullOrEmpty(expand))
+                {
+                    // If query string was empty, we add a new query string e.g ?$expand=Order. If not, we prepend a $expand query
+                    expand = queryStringIsEmpty ? "?" + expand : "?" + expand + "&" + queryString.TrimStart('?');
```

Comment:
Is there a test that tests for this? Like when we have a query string and when we dont?